src/hb-ot-var-avar-table.hh

```diff
@@ -89,10 +89,21 @@ struct SegmentMaps : ArrayOf<AxisValueMap>
     if (unlikely (arrayZ[i-1].fromCoord == arrayZ[i].fromCoord))
       return arrayZ[i-1].toCoord;
 
+#ifndef __has_builtin
+# define __has_builtin(x) 0
+#endif
+#if __has_builtin(__builtin_mul_overflow)
+# define hb_mul_overflow(x, y, result) __builtin_mul_overflow(x, y, &result)
+#else
+# define hb_mul_overflow(x, y, result) (result = (x) * (y), false)
+#endif
+    int factor;
+    if (hb_mul_overflow (arrayZ[i].toCoord - arrayZ[i-1].toCoord, value - arrayZ[i-1].fromCoord, factor))
+      return arrayZ[i-1].toCoord;
+#undef hb_mul_overflow
+
     int denom = arrayZ[i].fromCoord - arrayZ[i-1].fromCoord;
-    return arrayZ[i-1].toCoord +
-	   ((arrayZ[i].toCoord - arrayZ[i-1].toCoord) *
```

Comment:
Alternatively we can go for a line patch by adding a cast to `(long)` here which feels like moving the overflow which guess isn't that appropriate.